provider/maas/environ.go

```diff
@@ -1183,10 +1183,13 @@ func (environ *maasEnviron) waitForNodeDeployment2(id instance.Id, timeout time.
 		Total: timeout,
 	}
 
+	retryCount := 1
 	for a := longAttempt.Start(); a.Next(); {
 		machine, err := environ.getInstance(id)
 		if err != nil {
-			return errors.Trace(err)
+			logger.Warningf("failed to get instance from provider attempt %d", retryCount)
+			retryCount++
+			continue
 		}
 		stat := machine.Status()
```

Comment:
Maybe the rest of this loop should be moved outside of the loop. Once we get a provisioning error we should probably not retry so there is no need to have the rest of the body of this loop inside the loop. Agree?

Reply:
You could move it out and break the loop when err == nil.
The other paths all return though, so it's logically the same.